api/grpc_server.go

```diff
@@ -21,15 +22,74 @@ import (
 
 // SpacemeshGrpcService is a grpc server providing the Spacemesh api
 type SpacemeshGrpcService struct {
-	Server   *grpc.Server
-	Port     uint
-	StateApi StateAPI
-	Network  NetworkAPI
-	Tx       TxAPI
-	Mining   MiningAPI
-	Oracle   OracleAPI
-	GenTime  GenesisTimeAPI
-	Logging  LoggingAPI
+	Server        *grpc.Server
+	Port          uint
+	StateApi      StateAPI         // State DB
+	Network       NetworkAPI       // P2P Swarm
+	Tx            TxAPI            // Mesh
+	TxMempool     *miner.TxMempool // TX Mempool
+	Mining        MiningAPI        // ATX Builder
+	Oracle        OracleAPI
+	GenTime       GenesisTimeAPI
+	LayerDuration time.Duration
+	Logging       LoggingAPI
+}
+
+func (s SpacemeshGrpcService) getTransaction(txId types.TransactionId) (*types.Transaction, *types.LayerID, pb.TxStatus, error) {
```

Comment:
I'd revise to getTransactionStatue since this function return more than a simple tx